src/extension/csw/core/src/main/java/org/geoserver/csw/store/internal/CatalogStoreFeatureIterator.java

```diff
@@ -94,107 +79,46 @@ public CatalogStoreFeatureIterator(
             RecordDescriptor outputRecordDescriptor,
             Map<String, String> interpolationProperties) {
         this.interpolationProperties = interpolationProperties;
-        this.offset = offset;
-        this.count = count;
         this.sortOrder = sortOrder;
         this.filter = filter;
         catalogFacade = catalog.getFacade();
         this.mapping = mapping;
 
-        Filter advertised = ff.equals(ff.property("advertised"), ff.literal(true));
-
-        layerIt =
-                catalogFacade.list(
-                        ResourceInfo.class, ff.and(filter, advertised), null, null, sortOrder);
+        it = catalogFacade.list(PublishedInfo.class, filter, offset, count, sortOrder);
```

Comment:
I know it was already using the CatalogFacade but, this seems like a grave issue... dodging the catalog means no security is applied, anyone can see every layer regardless of what the security subsystem says? 
The catalog offers a similar list method, it's the one that should be used instead (and I believe that would also account for "advertised", along with workspace specific services and the like).

Reply:
That is a good point. However, it seems, strangely enough, that the Catalog list method only allows a single SortOrder, while the facade method allows multiple. CSW requires multiple sorts. Fixing this security leak would therefore require a catalog API change. ouch :/

Reply:
Hmm.. never mind, it seems that what you get there is a SecuredCatalogFacade, which is applying security. Never mind this one, but please check the other two feedbacks.

Reply:
Ok. I see the securecatalog does have a list with multi-sortby method to support this. It should ideally be added to the catalog interface as well.